### PR TITLE
Creating forms

### PR DESCRIPTION
**Pull request by Graycemuthui**
### Description
In this project I achieved the following:
- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
- It will return the first user from the database.
- Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
- Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
- Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### Changes made
- Add new method for the comments
- Fix routes

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    @comment = current_user.comments.new(comment_params)
+    @comment.post_id = params[:post_id]
+
+    if @comment.save
+      redirect_to user_post_path(current_user, @comment.post)
+    else
+      render :create
+    end
+  end
+
```
Comment (danmainah) [inline app/controllers/comments_controller.rb:12]: - [ ] Kindly add a new method in your comments controller as done in the posts controller, it's a project requirement. Also, ensure it has a corresponding router in the post controller routes.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Graycemuthui , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used


# Required Changes ♻️

- [ ] Kindly ensure your pages work. I couldn't get to view your forms in a browser, it showed a blank page even after adding users to the database.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (Graycemuthui) [reply app/controllers/comments_controller.rb:12]: Hey but the forms are just for the posts and comments 
To view the forms the route is http://127.0.0.1:3000/users/1/posts/new

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
+  def create
+    @comment = current_user.comments.new(comment_params)
+    @comment.post_id = params[:post_id]
+
+    if @comment.save
+      redirect_to user_post_path(current_user, @comment.post)
+    else
+      render :create
+    end
```
Comment (DeliceLydia) [inline app/controllers/comments_controller.rb:10]: - [ ] I am not able to add a comment please work on that and remember to add a route:
` /posts/:post_id/comments/new`

File: app/controllers/likes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  def create
+    @like = current_user.likes.new
+    @like.post_id = params[:post_id]
+    if @like.save
+      redirect_to user_post_path(current_user, @like.post)
+    else
+      render :create
+    end
+  end
+end
```
Comment (DeliceLydia) [inline app/controllers/likes_controller.rb:11]: - [ ] Currently, I am not able to add a `like` on the post please work on that

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Graycemuthui, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Graycemuthui 

Your project is complete! Great job for making the requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- post can be created
- likes and comments added
- routes ok
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**